kin/client.py

```diff
@@ -213,31 +213,33 @@ def kin_account(self, seed: str, channel_secret_keys: Optional[List[str]] = None
 
         tx_list = []
 
-        requested_amount = amount if amount < MAX_RECORDS_PER_REQUEST else MAX_RECORDS_PER_REQUEST
-
-        horizon_response = await self.horizon.account_transactions(address,
-                                                             cursor=cursor, limit=requested_amount,
-                                                             order='desc' if descending else 'asc')
-
-        for transaction in horizon_response['_embedded']['records']:
-            raw_tx = RawTransaction(transaction)
-            if simple:
-                try:
-                    simple_tx = SimplifiedTransaction(raw_tx)
-                    tx_list.append(simple_tx)
-                except KinErrors.CantSimplifyError:
-                    pass
-            else:
-                tx_list.append(raw_tx)
-            last_cursor = transaction['paging_token']
-
-        remaining_txs = amount - len(tx_list)
-        # if we got all the txs that we wanted, or there are no more txs
-        # TODO: paging does not work DP-370
-        if remaining_txs <= 0 or len(horizon_response['_embedded']['records']) < amount:
-            return tx_list
-        # If there are anymore transactions, recursively get the next transaction page
-        return tx_list.extend(await self.get_account_tx_history(address, remaining_txs, descending, last_cursor, simple))
+        while True:
+            requested_amount = min(amount, MAX_RECORDS_PER_REQUEST)
+
+            horizon_response = await self.horizon.account_transactions(address,
+                                                                       cursor=cursor, limit=requested_amount,
+                                                                       order='desc' if descending else 'asc')
+
+            current_loop_txs = []
+            for transaction in horizon_response['_embedded']['records']:
+                raw_tx = RawTransaction(transaction)
+                if simple:
+                    try:
+                        simple_tx = SimplifiedTransaction(raw_tx)
+                        current_loop_txs.append(simple_tx)
+                    except KinErrors.CantSimplifyError:
+                        pass
```

Comment:
hm, why not append it as is (like you did in the `else` section)?

Reply:
That's just the way the method works, if you only want simple txs it skips everything else